middleware/rest_api.py

```
@@ -289,3 +300,71 @@ def confirm_deduction(uid: str) -> DeductionConfirmResponse:
+
+        # Atomic write — temp file + rename
+        tmp_path = CONFIG_PATH + ".tmp"
+        with open(tmp_path, "w") as f:
+            yaml.dump(existing, f, default_flow_style=False, sort_keys=False)
+        os.replace(tmp_path, CONFIG_PATH)
+        logger.info("Web panel: config saved to %s", CONFIG_PATH)
+
+        # Restart the middleware service
+        try:
+            subprocess.Popen(
+                ["sudo", "systemctl", "restart", "spoolsense"],
```

Ruff (0.15.9)

[error] Starting a process with a partial executable path

(S607)